src_features/signTx/cmd_signTx.c

```diff
@@ -12,43 +12,37 @@ void handleSign(uint8_t p1,
                 unsigned int *tx) {
     UNUSED(tx);
     parserStatus_e txResult;
-    uint32_t i;
+    bip32_path_t bip32;
 
     if (os_global_pin_is_validated() != BOLOS_UX_OK) {
         PRINTF("Device is PIN-locked");
         THROW(0x6982);
     }
     if (p1 == P1_FIRST) {
-        if (dataLength < 1) {
-            PRINTF("Invalid data\n");
-            THROW(0x6a80);
-        }
         if (appState != APP_STATE_IDLE) {
             reset_app_context();
         }
         appState = APP_STATE_SIGNING_TX;
-        tmpCtx.transactionContext.pathLength = workBuffer[0];
-        if ((tmpCtx.transactionContext.pathLength < 0x01) ||
-            (tmpCtx.transactionContext.pathLength > MAX_BIP32_PATH)) {
-            PRINTF("Invalid path\n");
+
+        workBuffer = parseBip32(workBuffer, &dataLength, &bip32);
+
+        if (workBuffer == NULL) {
             THROW(0x6a80);
         }
-        workBuffer++;
-        dataLength--;
-        for (i = 0; i < tmpCtx.transactionContext.pathLength; i++) {
-            if (dataLength < 4) {
-                PRINTF("Invalid data\n");
-                THROW(0x6a80);
-            }
-            tmpCtx.transactionContext.bip32Path[i] = U4BE(workBuffer, 0);
-            workBuffer += 4;
-            dataLength -= 4;
-        }
+
+        tmpCtx.transactionContext.pathLength = bip32.length;
+        memcpy(tmpCtx.transactionContext.bip32Path, bip32.path, bip32.length * sizeof(uint32_t));
```

Comment:
with the new changes we can have situations like this

Reply:
Fixed in https://github.com/LedgerHQ/app-ethereum/pull/321/commits/e45e92ef6c0ea206562c147a153e2436adaa8d4e.